src/nvim/api/ui.c

```diff
@@ -275,18 +275,27 @@ static void push_call(UI *ui, const char *name, Array args)
         && strcmp(kv_A(temp_call, 0).data.string.data, "win_position") == 0) {
       Array call_data;
       int index;
-      if (kv_A(temp_call, 1).type == kObjectTypeArray) {
-        call_data = kv_A(temp_call, 1).data.array;
-        index = 1;
-      } else {
-        call_data = temp_call;
-        index = 2;
+      int did = 0;
+      // the adjacent calls are bundled together into one call so we must
+      // go through all the calls to find and update the relevant calls.
+      for (int i = 1; i < kv_size(temp_call); i++) {
```

Comment:
@bfredl This achieves what has to be done, but looks a little too complicated(?) Any suggestion about how we can make this not look ugly? :sweat_smile: Or is it fine? :thinking:

Reply:
Where are these events generated in the first place in `window.c`? How tricky would it be to throttle them there?